pandas/core/generic.py

```diff
@@ -8564,6 +8523,179 @@ def nanptp(values, axis=0, skipna=True):
             ``numpy.ndarray`` method ``ptp``.""",
             nanptp)
 
+    _shared_docs['transform'] = ("""
+    Apply a function that returns a same-shaped object.
+
+    .. versionadded:: 0.20.0
+
+    Parameters
+    ----------
+    func : callable, list of callables, or dict
+        The function or function to apply to the DataFrame or Series.
+        Each function should expect a Series and return an array-like
+        with the same shape. Accepted combinations are:
+
+        - function : Applied to the Series, or to each column of the
+          DataFrame independently.
+        - list of functions : Each function is applied to the Series
+          or to each column of the DataFrame independently.
+        - dict of column names -> functions or list of functions :
+          For DataFrames only, apply the function or functions for a
+          given key to a specific column.
+
+    axis : {0 or 'index', 1 or 'columns'}, default 'index'
+        Axis to apply the function along (DataFrame only).
+
+        * 0 or 'index' : apply along the index, i.e. the function
+          gets each column.
+        * 1 or 'columns' : apply along the columns, i.e. the function
+          gets each row.
+
+    *args
+        Additional positional arguments are passed to `func`.
+    **kwargs
+        Additional keyword arguments are passed to `func`.
+
+    Returns
+    -------
+    transformed : Series or DataFrame
+        The result type is the same as the input. Furthermore, it is
+        the same shape.
+
+    Notes
+    ------
+    The function performs the transformation.
+    It takes a Series as input and must return a same length Series.
+    Note that the function may modify the Series index.
+
+    If applied to a Series the function is applied once.
+    If applied to a DataFrame the function is applied on each row or column.
+    Note that if the function modifies the index it must do so consistently
+    in the case of a DataFrame.
+
+    See also
+    --------
+    pandas.DataFrame.aggregate : Perform aggregate type of operations.
+    pandas.DataFrame.apply : Apply general functions.
+
+    Examples
+    --------
+
+    transform() is applicable to Dataframes.
+
+    >>> df = pd.DataFrame(np.arange(15).reshape(5, 3), columns=['A', 'B', 'C'],
+    ...                   index=pd.date_range('1/1/2000', periods=5))
+    ...
+    >>> df
+                A   B   C
+    2000-01-01   0   1   2
+    2000-01-02   3   4   5
+    2000-01-03   6   7   8
+    2000-01-04   9  10  11
+    2000-01-05  12  13  14
+
+    By default, `func` is applied to each column independently.
+
+    >>> df.transform(lambda x: (x - x.mean()) / x.std())
```

Comment:
I'd try to find a more real-world example. I think for beginners it's a bit difficult to follow with meaningless data (A, B, C...) and lambdas.

Also, you may want to take a look at this document: http://pbpython.com/pandas_transform.html (or some other on how transform is usually used).

Reply:
Frankly I disagree.  
(1) There is no clearer column names than A, B, C (Excel like) (2) the lambda example is taken from current doc about transform (3) I wrote other examples without lambda (4) at this stage of the example it is just to show it is applied by column and it gives the same result because the data is just shifted from one column to the next. (5) you're the 3rd person (legitimately) giving opinion about the examples and (naturally) each has a different one, often in contradiction... it going in circles...

The page you mention mixes groupby and transform so in my view is more complex and not what I, as a (heavy) user, would expect in the basic doc.

Reply:
Just to be clear.

I'll understand perfectly if you prefer to drop the PR - if only because there are some involuntary code change (bloody IDE !) - or the examples do not suit you.

Please then just let me know so I can close it.

Pandas is great and I'm a heavy almost daily user - and increasingly more people in my company. Congratulations !